pkg/controller/bootstrap/tokencleaner.go

```diff
@@ -71,13 +69,15 @@ type TokenCleaner struct {
 }
 
 // NewTokenCleaner returns a new *NewTokenCleaner.
-func NewTokenCleaner(cl clientset.Interface, secrets coreinformers.SecretInformer, options TokenCleanerOptions) (*TokenCleaner, error) {
+func NewTokenCleaner(cl clientset.Interface, options TokenCleanerOptions) (*TokenCleaner, error) {
+	informerFactory := informers.NewSharedInformerFactoryWithOptions(cl, options.SecretResync, informers.WithNamespace(options.TokenSecretNamespace))
```

Comment:
I don't spinning up an entire informer separate from the other informer can improve performance over sharing an informer with the other controllers that watch secrets. On the contrary, this looks to me like it will add a bunch of overhead.